lib/live_debugger_refactor/app/utils/term_parser.ex

```diff
@@ -4,14 +4,36 @@ defmodule LiveDebuggerRefactor.App.Utils.TermParser do
   Based on [Kino.Tree](https://github.com/livebook-dev/kino/blob/main/lib/kino/tree.ex)
   """
 
-  @type display_element() :: %{text: String.t(), color: String.t() | nil}
-  @type tree_node() :: %{
-          kind: String.t(),
-          children: [tree_node()] | nil,
-          content: [display_element()],
-          expanded_before: [display_element()] | nil,
-          expanded_after: [display_element()] | nil
-        }
+  defmodule DisplayElement do
+    @moduledoc false
+    defstruct [:text, color: nil]
+
+    @type t :: %__MODULE__{
+            text: String.t(),
+            color: String.t() | nil
+          }
+  end
+
+  defmodule TermNode do
+    @moduledoc """
+    Represents a node in the display tree.
+
+    - `kind`: The type of the node (e.g., "atom", "list", "map").
+    - `children`: A list of child nodes, if any.
+    - `content`: Display elements that represent the content of the node when has no children or not expanded.
+    - `expanded_before`: Display elements shown before the node's children when expanded.
+    - `expanded_after`: Display elements shown after the node's children when expanded.
+    """
+    defstruct [:kind, :children, :content, :expanded_before, :expanded_after]
+
+    @type t :: %__MODULE__{
+            kind: String.t(),
+            children: [TermNode.t()] | nil,
```

Comment:
Wouldn't it make more sense if these values could be just list and if there are no `TermNode` the list is empty?